### PR TITLE
[E118-MG < T0974-MG] Remove multi-threaded related logic and variables

### PR DESCRIPTION
**Pull request by antaljanosbenjamin**
[master < Epic] PR
- [ ] Check, and update documentation if necessary
- [ ] Update [changelog](https://docs.memgraph.com/memgraph/changelog)
- [ ] Write E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch and the Epic branch
- [ ] Provide the full content or a guide for the final git message

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Update [changelog](https://docs.memgraph.com/memgraph/changelog)
- [x] Provide the full content or a guide for the final git message

> * Remove logic that was necessary for optimal multi-threaded performance, such
>  as accumulating deleted object in local containers and appending them to a
>  global one, handling overlapping locking.
> * Remove background GC thread.
> * Remove mutexes, locks and atomics throughout storage.

### CODE REVIEW

File: src/storage/v2/storage.cpp [hunk 1]
```diff
@@ -1110,8 +1110,9 @@ void Storage::Accessor::Abort() {
   {
     std::unique_lock<utils::SpinLock> engine_guard(storage_->engine_lock_);
     uint64_t mark_timestamp = storage_->timestamp_;
-    // Take garbage_undo_buffers lock while holding the engine lock to make
-    // sure that entries are sorted by mark timestamp in the list.
+    // Take garbage_undo_buffers lock while holding the engine lock to make sure that entries are sorted by mark
+    // timestamp in the list. This is necessary when a transaction is aborting simultaneously with a GC run: boht of
```
Comment (jbajic) [inline src/storage/v2/storage.cpp:1114]:
```suggestion
    // timestamp in the list. This is necessary when a transaction is aborting simultaneously with a GC run: both of
```

File: src/storage/v3/commit_log.cpp [hunk 2]
```diff
@@ -47,19 +47,14 @@ CommitLog::~CommitLog() {
...
   if (id == oldest_active_) {
     UpdateOldestActive();
   }
 }
 
-uint64_t CommitLog::OldestActive() {
-  std::lock_guard<utils::SpinLock> guard(lock_);
-  return oldest_active_;
-}
+uint64_t CommitLog::OldestActive() const { return oldest_active_; }
```
Comment (jbajic) [inline src/storage/v3/commit_log.cpp:57]: You can also mark it as `noexcept`

**Review by jbajic — APPROVED** [review]

File: src/storage/v3/storage.cpp [hunk 3]
```diff
@@ -401,19 +401,17 @@ Storage::Storage(Config config)
...
-            spdlog::warn(
-                utils::MessageWithLink("Snapshots are disabled for replicas.", "https://memgr.ph/replication"));
-            break;
-        }
-      }
-    });
-  }
-  if (config_.gc.type == Config::Gc::Type::PERIODIC) {
-    gc_runner_.Run("Storage GC", config_.gc.interval, [this] { this->CollectGarbage<false>(); });
+    // TODO(antaljanosbenjamin): handle snapshots
```
Comment (42jeremy) [inline src/storage/v3/storage.cpp:404]: Is it TODO for now? If not, feel free to resolve it to close the PR

File: src/storage/v3/storage.cpp [hunk 4]
```diff
@@ -437,7 +432,7 @@ Storage::~Storage() {
     wal_file_ = std::nullopt;
   }
   if (config_.durability.snapshot_wal_mode != Config::Durability::SnapshotWalMode::DISABLED) {
-    snapshot_runner_.Stop();
+    // TODO(antaljanosbenjamin): stop snapshot creation
```
Comment (42jeremy) [inline src/storage/v3/storage.cpp:435]: Same than https://github.com/memgraph/memgraph/pull/460/files#r957515437

**Review by 42jeremy — APPROVED** [review]

File: src/storage/v3/storage.cpp (hunk 3 above)
Comment (antaljanosbenjamin) [reply src/storage/v3/storage.cpp:404]: The purpose of TODOs is keeping the more accurate knowledge about what should be done where etc. We have a lot of them as a big part of the storage won't be working in M1.

**Pull request merged**